libs/database/src/lib/entities/polygon-geometry.entity.ts

```diff
@@ -22,4 +23,88 @@ export class PolygonGeometry extends Model<PolygonGeometry> {
   @AllowNull
   @Column(BIGINT.UNSIGNED)
   createdBy: number | null;
+
+  static async checkIsSimple(polygonUuid: string): Promise<boolean> {
+    if (this.sequelize == null) {
+      throw new InternalServerErrorException("PolygonGeometry model is missing sequelize connection");
+    }
+
+    const result = (await this.sequelize.query(
+      `
+        SELECT ST_IsSimple(geom) as isSimple
+        FROM polygon_geometry 
+        WHERE uuid = :polygonUuid
+      `,
+      {
+        replacements: { polygonUuid },
+        type: QueryTypes.SELECT
+      }
+    )) as { isSimple: boolean }[];
+
+    if (result.length === 0) {
+      throw new NotFoundException();
+    }
```

Comment:
I don't think `NotFoundException` is reasonable to throw in the model files - there might be uses of this function down the road that would rather not bail immediately if it's not found. Instead, please `return result[0]?.isSimple` and check for a null response in the validator, throwing the NotFoundException there instead. This method signature will need to change to `Promise<boolean | undefined>`